tests.js

```diff
@@ -6,31 +6,80 @@ const expect = require('chai').expect;
 const easyFix = require('./index');
 
 const ASYNC_DELAY = 1000;
-const thingToTest = {
+
+function createThingToTest() {
```

Comment:
Instead of restructuring these tests, we can simply add new tests (for Promises) defined in the runSharedTests method.  These will get picked up by the `describe` definitions for the different test modes below.